Update with style guide (#1133)
iotaledger/hornet: docs/post_installation/peering.md

# Peering Configuration

> The IOTA network is a distributed network. It uses a gossip protocol to broadcast data among IOTA nodes. To participate in a network, each node has to establish a secure connection to other nodes in the network(to its peer neighbors), and mutually exchange messages.

Reviewer: ```suggestion
The IOTA network is a distributed network. It uses a gossip protocol to broadcast data among IOTA nodes. To participate in a network, each node has to establish a secure connection to other nodes in the network (to its peer neighbors), and mutually exchange messages.
```

2021-04-19T14:27:55Z  INFO    P2P     peer configured, ID: 12D3KooWEWunsQWGvSWYN2VR7wNNoHgae4XikBqwSre8K8sVTefu
```

Your `PeerId` is an essential part of your `multiaddr` used to configure neighbors. For example,  `/dns/example.com/tcp/15600/p2p/12D3KooWHiPg9gzmy1cbTFAUekyLHQKQKvsKmhzB7NJ5xnhK4WKq`, where `12D3KooWHiPg9gzmy1cbTFAUekyLHQKQKvsKmhzB7NJ5xnhK4WKq` corresponds to your `PeerId`. Your `PeerId` is also visible on the start page of the dashboard.

> However, we recommended that you pre-generate the identity.  This way you can pre-communicate it to your peers before you even start your node, and you will be able to retain the identity in case you accidentally delete your `./p2pstore`.

Reviewer: ```suggestion
However, we recommended that you pre-generate the identity. This way you can pre-communicate it to your peers before you even start your node, and you will be able to retain the identity in case you accidentally delete your `./p2pstore`.
```

Your p2p PeerID:  12D3KooWH1vQ5SWtEUTVNCCCxxgGeoLZLVz1PnpCcjhcJXcGB9cu
```

Now, copy the value of `Your p2p private key` to the `p2p.identityPrivateKey` configuration option.

Your Hornet node will now use the specified private key in `p2p.identityPrivateKey` to generate the `PeerId` (which will
ultimately be stored in `./p2pstore`).

:::info

> In case there already is a `./p2pstore` with another identity, Hornet will panic and tell you that you have a previous identity which does not match with what is defined via `p2p.identityPrivateKey` .  In this case, you should either delete the `./p2pstore` or reset the `p2p.identityPrivateKey`.

Reviewer: ```suggestion
In case there already is a `./p2pstore` with another identity, Hornet will alert you that you have a previous identity which does not match with what is defined via `p2p.identityPrivateKey` .  In this case, you should either delete the `./p2pstore` or reset the `p2p.identityPrivateKey`.
```

For example, a node is reachable using IPv4 `100.1.1.1` using `TCP` on port `15600` and its `PeerId`
is `12D3KooWHjcCgWPnUEP8wNdbL2fx63Cmosk16xyZ25iUZagxmHb4`. A `multiaddr` encoding of this information would look like this:

```plaintext
/ip4/100.1.1.1/tcp/15600/p2p/12D3KooWHjcCgWPnUEP8wNdbL2fx63Cmosk16xyZ25iUZagxmHb4
```

:::info
Note how `ip4` is used. A common mistake is to use `ipv4`.
:::

If a node is reachable using a DNS name (for example `node01.iota.org`), then the given `multiaddr` would be:

```plaintext
/dns/node01.iota.org/tcp/15600/p2p/12D3KooWHjcCgWPnUEP8wNdbL2fx63Cmosk16xyZ25iUZagxmHb4
```

You will need to find out your own `multiaddr` to give to your peers for neighboring. To do so, combine the `peerId` you have gotten
from the stdout when the Hornet node started up (or which was shown via the `p2pidentity` CLI tool), and your
configured `p2p.bindAddress`. Replace the `/ip4/<ip_address>`/`/dns/<hostname>` segments with the actual information.

> You can find more information about `multiaddr` is available at the [libp2p docs page](https://docs.libp2p.io/concepts/addressing/).

Reviewer: ```suggestion
You can find more information about `multiaddr` at the [libp2p docs page](https://docs.libp2p.io/concepts/addressing/).
```

## Adding Node Peers

> Once you know your node's own `multiaddr`, it can be exchanged with other node owners to establish a mutual peer connection. We recommended number of peer neighbors between 4-6 to get some degree of redundancy.

Reviewer: ```suggestion
Once you know your node's own `multiaddr`, it can be exchanged with other node owners to establish a mutual peer connection. We recommended a number of peer neighbors between 4-6 to get some degree of redundancy.
```

> ## Finding neighbors

Reviewer: ```suggestion
## Finding Neighbors
```

}
```

## Autopeering

> Hornet also supports automatically finding peers through the _autopeering_ module.  To minimize service distribution in case your autopeered peers are flaky, we recommend to only use autopeering if you have at least 4 static peers.

Reviewer: ```suggestion
Hornet also supports automatically finding peers through the _autopeering_ module. To minimize service distribution in case your autopeered peers are flaky, we recommend to only use autopeering if you have at least 4 static peers.
```

Autopeering is enabled by default,  and your node will use the specified entry nodes under `p2p.autopeering.entryNodes` to find new peers. `entryNodes` are also encoded as `multiaddr`:

```
/ip4/45.12.34.43/udp/14626/autopeering/8CZELJwB3aBzxJgnLMvvt1FirAwNN6jif9LavYTNHCty
```

Where the `/autopeering` portion defines the base58 encoded Ed25519 public key.

By default, Hornet will peer up to 4 autopeered peers and initiate a gossip protocol with them. Autopeered peers are not subject to connection trimming, the same way as mutually tethered peers aren't either.

> ### Entry node

Reviewer: ```suggestion
### Entry Node
```

If you want to run your own node as an autopeering entry node, you should enable `p2p.autopeering.runAsEntryNode`. The base58 encoded public key is in the output of the `p2pidentity` Hornet tool. Alternatively, if you already have an identity in a `./p2pstore`, you can use the `p2pidentityextract` Hornet tool to extract it.

### Low/High Watermark

> The `p2p.connectionManager.highWatermark` and `p2p.connectionManager.lowWatermark` configuration options define "watermark" points.  Watermark points can be thought of as a filling basin where if the `highWatermark` is reached, water will be drained until it reaches the `lowWatermark` again. Similarly, the connection manager within Hornet will start trimming away connections to peers if `highWatermark` peers are connected until it reaches `lowWatermark` count of peers. These watermarks exist for a certain buffer number of peers to be connected, which will not necessarily  be targeted by the gossip protocol.

Reviewer: ```suggestion
The `p2p.connectionManager.highWatermark` and `p2p.connectionManager.lowWatermark` configuration options define "watermark" points.  Watermark points can be thought of as a filling basin where if the `highWatermark` is reached, water will be drained until it reaches the `lowWatermark` again. Similarly, the connection manager within Hornet will start trimming away connections to peers if `highWatermark` peers are connected until it reaches `lowWatermark` count of peers. These watermarks exist for a certain buffer number of peers to be connected, which will not necessarily be targeted by the gossip protocol.
```
